### PR TITLE
enable token to token trade on kyber network

### PR DESCRIPTION
**Pull request by ilanDoron**
please don't pull
only added this PR so its easier to see differences

### CODE REVIEW

File: contracts/KyberNetwork.sol [hunk 1]
```diff
@@ -40,8 +40,8 @@ contract KyberNetwork is Withdrawable, Utils {
     }
     /* solhint-enable no-complex-fallback */
 
-    event ExecuteTrade(address indexed sender, ERC20 src, ERC20 dest, uint actualSrcAmount, uint actualDestAmount);
-
+    event expected(uint actualDest, uint expectedDest);
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: never used. and bad convention name.

File: contracts/KyberNetwork.sol [hunk 2]
```diff
@@ -90,8 +90,8 @@ contract KyberNetwork is Withdrawable, Utils {
         userSrcBalanceAfter = getBalance(src, msg.sender);
         userDestBalanceAfter = getBalance(dest, destAddress);
 
-        require(userSrcBalanceAfter <= userSrcBalanceBefore);
-        require(userDestBalanceAfter >= userDestBalanceBefore);
+        require(userSrcBalanceAfter < userSrcBalanceBefore);
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: why strict inequality?

File: contracts/KyberNetwork.sol [hunk 3]
```diff
@@ -214,11 +214,39 @@ contract KyberNetwork is Withdrawable, Utils {
     /// @dev best conversion rate for a pair of tokens, if number of reserves have small differences. randomize
     /// @param src Src token
     /// @param dest Destination token
+    function findBestRate(ERC20 src, ERC20 dest, uint srcQty) public view returns(uint noUse, uint rate) {
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: why return param is called `noUse`?

File: contracts/KyberNetwork.sol [hunk 4]
```diff
@@ -214,11 +214,39 @@ contract KyberNetwork is Withdrawable, Utils {
...
+
+        (reserve2, rateEthToDest) = searchBestRate(ETH_TOKEN_ADDRESS, dest, ethAmount);
+        actualDest = calcDestAmount(ETH_TOKEN_ADDRESS, dest, ethAmount, rateEthToDest);
+
+        if (rateSrcToEth == PRECISION) rate = rateEthToDest;
+        else if (rateEthToDest == PRECISION) rate = rateSrcToEth;
+        else rate = calcRateFromQty(srcQty, actualDest, getDecimals(src), getDecimals(dest));
+    }
+
     /* solhint-disable code-complexity */
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: you can probably remove it now right?

File: contracts/KyberNetwork.sol [hunk 5]
```diff
@@ -356,6 +403,13 @@ contract KyberNetwork is Withdrawable, Utils {
     {
         uint callValue = 0;
 
+        if (src == dest) {
+            //this is for a "fake" trade when both src and dest are ethers.
+            if (destAddress != (address(this)))
```
Comment (yaronvel) [inline contracts/KyberNetwork.sol]: is this good? looks bad.

File: contracts/KyberNetwork.sol (hunk 2 above)
Comment (ilanDoron) [reply contracts/KyberNetwork.sol]: because it can't remain the same. if its same it means no trade was done. better revert.

**Pull request merged**